crates/sui/src/unit_tests/snapshots/sui__upgrade_compatibility__upgrade_compatibility_tests__function.snap

```diff
@@ -56,5 +56,50 @@ error[Compatibility E03001]: function signature mismatch
    = Functions are part of a module's public interface and cannot be changed during an upgrade.
    = Restore the original function's return types for function 'func_with_wrong_return_length'.
 
+error[Compatibility E03001]: function signature mismatch
+   ┌─ /fixtures/upgrade_errors/function_errors_v2/sources/UpgradeErrors.move:42:45
+   │
+42 │     public fun func_with_wrong_struct_param(a: StructB): u64 {
+   │                                             ^ Unexpected parameter '0x0::upgrades::StructB', expected '0x0::upgrades::StructA'
```

Comment:
maybe? 
```suggestion
   │                                             ^ Unexpected parameter with type '0x0::upgrades::StructB', expected '0x0::upgrades::StructA'
```

Reply:
+1 to this -- in the binary representation, all we see is a type, but we should clarify for the sake of people reading source code.